speculos/mcu/ocr.py

```diff
@@ -159,22 +159,44 @@ def find_char_from_bitmap(bitmap: BitMap):
     'c'
     """
     all_values = []
+    legacy = True
     for font in bagl_font.FONTS:
         font_map = get_font_map(font)
-        for character_value, bitmap_struct in font_map.items():
-            if bitmap_struct.bitmap == bitmap:
-                all_values.append(character_value)
+        if font_map in OCR.json_fonts:
+            # This is a loaded JSON font => new behaviour
+            for character_value, bitmap_struct in font_map.items():
+                if bitmap_struct.bitmap == bitmap:
+                    all_values.append(character_value)
+                    legacy = False
```

Comment:
I do not agree with the **break** suggestion.
There was no break before, and I don't see why I should add one.
This will remove the use of _all_value_ list.
To be honest, I don't really undestand what mean this code:
```
if all_values:
    char = max([x for x in all_values])
```
If we decide this code is useless, its not a **break** which should be done, but a **return**

But this code is not useless, I already saw it replace a incorrect 'I' by a 'l' (uppercase 'i' by a lowercase 'L'), which bitmap content was identical.
(but, again, I have to admit I don't really understand how it works and I doubt it will always find the _correct answer_, `max([x for x in all_values])` can't work everytime...)
(we can try to find a better solution, but its not in the scope of this PR)